.github/workflows/gpuci.yml

```diff
@@ -0,0 +1,43 @@
+name: "GPU CI/CD"
+
+on:
+  pull_request:
+    branches:
+      - 'main'
+      - 'r**'
+    types: [ labeled ]
+
+jobs:
+  gpu-test:
+    runs-on: self-hosted
+    if: ${{ github.event.label.name == 'gpuci' }}
+
+    steps:
+    - name: Run nvidia-smi test
+      run: |
+        whoami
+        nvidia-smi
+
+    - name: Checkout the PR code
+      uses: actions/checkout@v3
+
+    - name: Set up Python 3.10
+      uses: actions/setup-python@v4
+      with:
+        python-version: '3.10'
+
+    - name: Install NeMo-Curator and PyTest
+      run: |
+        sudo apt-get update
+        sudo apt-get install -y build-essential cmake g++ python3-dev
```

Comment:
Maybe this works but something to be careful about is which `gcc` (or `g++`) it ends up using, since one doesn't work for fasttext==0.9.2 (https://github.com/facebookresearch/fastText/issues/1281)